### PR TITLE
fix: x-forwarded-port header is 'undefined' when no port in url

### PR DESCRIPTION
**Pull request by yuvalotem**
### What?
See this issue - https://github.com/vercel/next.js/issues/61133

following this change https://github.com/vercel/next.js/pull/57815 `x-forwarded-port` header value is 'undefined' if the URL has no port

### Why?
x-forwarded-port 'undefined' makes other http-proxy throw 405 error for the invalid header value

### How?
Give default 80 port if the URL has no port

### CODE REVIEW

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]
This is looking good @yuvalotem - can you come up with a test for it perhaps? You may be able to modify an existing one too. I will also look into this and then we can land your changes. Thank you!

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 7188a799428d214e08564b3879abd14b4ac70b61

`pnpm test-dev test/development/app-hmr/hmr.test.ts`
- app-dir-hmr > filesystem changes > should update server components pages when env files is changed (edge)

<details>
<summary>Expand output</summary>

  ● app-dir-hmr › filesystem changes › should update server components pages when env files is changed (edge)

    page.goto: Timeout 60000ms exceeded.
    Call log:
      - navigating to "http://localhost:44767/env/edge", waiting until "load"

      274 |     opts?.beforePageLoad?.(page)
      275 |
    > 276 |     await page.goto(url, { waitUntil: 'load' })
          |                ^
      277 |   }
      278 |
      279 |   back(options): BrowserInterface {

      at BrowserInterface.goto (lib/browsers/playwright.ts:276:16)
      at webdriver (lib/next-webdriver.ts:129:3)
      at Object.<anonymous> (development/app-hmr/hmr.test.ts:77:25)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by Ethan-Arrowood — APPROVED** [review]
I cannot determine a way to test this issue using a local Next.js server (even a custom one). 

I've created https://github.com/vercel/vercel/pull/11254 to test this e2e. I've asserted that the currently described behavior in https://github.com/vercel/next.js/issues/61133 is valid (see commented output https://github.com/vercel/vercel/pull/11254#issuecomment-1989332557).

Once this is landed, we can verify the new test passes and then land it to prevent regressions.

File: packages/next/src/server/base-server.ts [hunk 1]
```diff
@@ -889,13 +889,15 @@ export default abstract class Server<ServerOptions extends Options = Options> {
...
       }
 
-      req.headers['x-forwarded-host'] ??= req.headers['host'] ?? this.hostname
-      req.headers['x-forwarded-port'] ??= this.port?.toString()
       const { originalRequest } = req as NodeNextRequest
-      req.headers['x-forwarded-proto'] ??= (originalRequest.socket as TLSSocket)
-        ?.encrypted
-        ? 'https'
-        : 'http'
+      const isHttps = !!(originalRequest.socket as TLSSocket)?.encrypted
```
Comment (Ethan-Arrowood) [inline packages/next/src/server/base-server.ts:893]:
```suggestion
      const isHttps = !!(originalRequest?.socket as TLSSocket)?.encrypted
```

This should fix the broken tests.

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

**Review by Ethan-Arrowood — APPROVED** [review]

**Pull request merged**